rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/kustomize/KustomizeTemplateUtils.java

```diff
@@ -0,0 +1,138 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.rosco.manifests.kustomize;
+
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import com.netflix.spinnaker.rosco.jobs.BakeRecipe;
+import com.netflix.spinnaker.rosco.manifests.TemplateUtils;
+import com.netflix.spinnaker.rosco.manifests.kustomize.mapping.Kustomization;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+
+@Component
+@Slf4j
+public class KustomizeTemplateUtils extends TemplateUtils {
+
+  private HashSet<String> filesToDownload = new HashSet<String>();
+
+  @Autowired
+  private KustomizationFileReader kustomizationFileReader;
+
+  public BakeRecipe buildBakeRecipe(BakeManifestEnvironment env, KustomizeBakeManifestRequest request) {
+    BakeRecipe result = new BakeRecipe();
+    result.setName(request.getOutputName());
+    filesToDownload = new HashSet<>();
+    Artifact artifact = request.getInputArtifact();
+    if (artifact == null ) {
+      throw new IllegalArgumentException("At least one input artifact must be provided to bake");
+    }
+    String kustomizationFileName = FilenameUtils.getName(artifact.getReference());
+    if(kustomizationFileName!=null && !kustomizationFileName.toUpperCase().contains("KUSTOMIZATION")){
+      throw new IllegalArgumentException("The inputArtifact should be a kustomization file valid.");
+    }
+
+    List<Artifact> artifacts = new ArrayList<>();
+    String templatePath = getKustomizationPath(env,artifact);
+    try {
+      Path path = Paths.get(FilenameUtils.getPath(artifact.getReference()));
+        artifact.setReference(path.toString());
+      for (String s : getFilesFromGithub(artifact)) {
+        Artifact a = new Artifact();
+          a.setReference(s);
+          a.setArtifactAccount(artifact.getArtifactAccount());
+          a.setCustomKind(artifact.isCustomKind());
+          a.setLocation(artifact.getLocation());
+          a.setMetadata(artifact.getMetadata());
+          a.setName(artifact.getName());
+          a.setProvenance(artifact.getProvenance());
+          a.setType(artifact.getType());
+          a.setUuid(artifact.getUuid());
+          a.setVersion(artifact.getVersion());
+          artifacts.add(a);
+      }
```

Comment:
Small nit - A cleaner way to write this might be - 

```
HashSet<String> files = getFilesFromGithub(artifact)
List<Artifact> artifacts = files.stream()
	.map((f) -> {
		return Artifact.builder()
		  .reference(f)
		  .artifactAccount(artifact.getArtifactAccount())
		  .customKind(artifact.isCustomKind())
		  .location(artifact.getLocation())
		  .metadata(artifact.getMetadata())
		  .name(artifact.getName())
		  .provenance(artifact.getProvenance())
		  .type(artifact.getType())
		  .version(artifact.getVersion())
	})
       .collect(Collectors.toList());
```

Reply:
Streaming operations are more common in the Spinnaker codebases so it fits well with the style of the rest of the code.

Reply:
changed